src/overlays/actors/ovl_Bg_Toki_Swd/z_bg_toki_swd.c

```diff
@@ -22,19 +314,174 @@ const ActorInit Bg_Toki_Swd_InitVars =
+static SubActor98Init subActor98Init =
+{
+    0x0A,
+    0x0023,
+    0x0064,
+    0xFF,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Toki_Swd/func_808BAF40.s")
+static InitChainEntry initChain[] =
+{
+    ICHAIN_VEC3F_DIV1000(scale, 0x19, ICHAIN_STOP),
+};
+
+static void BgTokiSwd_SetupAction(BgTokiSwd* this, ActorFunc actionFunc)
+{
+   this->actionFunc = actionFunc;
+}
+
+static void BgTokiSwd_Init(BgTokiSwd* this, GlobalContext* globalCtx) 
+{
+   ColliderCylinderMain* collision;
+   Actor* thisx = &this->actor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Toki_Swd/func_808BB0AC.s")
+   Actor_ProcessInitChain(thisx, initChain);
+   this->actor.shape.unk_08 = 800.0f;
+   BgTokiSwd_SetupAction(thisx, func_808BAF40);
+   collision = &this->collider;
+   if (LINK_IS_ADULT) 
+   { 
+       thisx->draw = NULL;
+   }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Toki_Swd/func_808BB128.s")
+   if (gSaveContext.scene_setup_index == 5) 
+   {
+       globalCtx->unk_11D30[0] = 0xFF;
+   }
+
+   ActorCollider_AllocCylinder(globalCtx, collision);
+   ActorCollider_InitCylinder(globalCtx, collision, thisx, &colliderInit);
+   ActorCollider_Cylinder_Update(thisx, collision);
+   func_80061ED4(&thisx->sub_98, 0, &subActor98Init);
+}
+
+static void BgTokiSwd_Destroy(BgTokiSwd* this, GlobalContext* globalCtx)
+{
+   ColliderCylinderMain* collider = &this->collider;
+   ActorCollider_FreeCylinder(globalCtx, collider);
+}
+
+static void func_808BAF40(BgTokiSwd* this, GlobalContext* globalCtx) 
+{
+   if (((gSaveContext.event_chk_inf[4] & 0x8000) == 0) && (gSaveContext.scene_setup_index < 4) && 
+        (func_8002E12C(&this->actor, 800.0f, 0x7530) != 0) && (func_800BFC84(globalCtx) == 0)) 
+   {
+       gSaveContext.event_chk_inf[4] |= 0x8000;
+       globalCtx->csCtx.segment = D_808BBD90;
+       gSaveContext.cutscene_trigger = 1;
+   } 
+   if (LINK_IS_CHILD || ((gSaveContext.event_chk_inf[5] & 0x20) != 0)) 
+   {
+       if (func_8002F410(&this->actor, globalCtx) != 0) 
+       {
+           if (LINK_IS_CHILD) 
+           {
+               Item_Give(globalCtx, 0x3C);
+               globalCtx->csCtx.segment = D_808BB2F0;
+           } 
+           else 
+           {
+               globalCtx->csCtx.segment = D_808BB7A0;
+           }
+           Audio_SetBGM(NA_BGM_STOP);
+           Audio_SetBGM(NA_SE_PL_BOUND_DIRT);
+           gSaveContext.cutscene_trigger = (u8)1;
+           this->actor.attachedA = 0;
+           BgTokiSwd_SetupAction(this, func_808BB0AC);
+       } 
+       else 
+       {
+           if (func_8002E084(&this->actor, 0x2000) != 0) 
+           {
+               func_8002F580(&this->actor, globalCtx);
+           }
+       }
+   } 
+   if (gSaveContext.scene_setup_index == 5) 
+   {
+       if (globalCtx->unk_11D30[0] > 0) 
+       {
+           globalCtx->unk_11D30[0] = globalCtx->unk_11D30[0] - 1;
```

Comment:
Does -- still match? If so I would use that. Or -= alternatively.
```suggestion
           globalCtx->unk_11D30[0]--;
```